testcontainers/src/core/client.rs

```diff
@@ -312,15 +312,15 @@ impl Client {
 
     pub(crate) async fn docker_hostname(&self) -> Result<url::Host, ClientError> {
         let docker_host = self.config.docker_host();
-        match docker_host.scheme() {
-            "tcp" | "http" | "https" => {
-                docker_host
-                    .host()
-                    .map(|host| host.to_owned())
-                    .ok_or_else(|| {
-                        ConfigurationError::InvalidDockerHost(docker_host.to_string()).into()
-                    })
-            }
+        let docker_host_url = Url::from_str(docker_host).expect("expected a valid Docker Host URL");
```

Comment:
This expectation was already occurring within `config.docker_host()`, FWIW.

This will not be a problem for `NamedPipeConnector`. The path always did parse without `Error`/panic, it's just that it would normalize to an erroneous Url. Parsing the Url remains fine so long as we don't pass it on to the clients directly. We still parse it here so we can differentiate the scheme, for example.

Reply:
That makes sense, but could you replace `expect` with an error?
We prefer not to panic on library level and user is responsible for handling.

It used to be 

```rs
|url::ParseError| ConfigurationError::InvalidDockerHost(e.to_string()))?;
```

Reply:
Maybe I'm missing something? It seems that `config.docker_host()` always panicked:

https://github.com/testcontainers/testcontainers-rs/blob/e354b7532b5d448ab37c9f95dab6c68c592cb62b/testcontainers/src/core/env/config.rs#L141

I agree with you though and I've gone ahead and made the change in e1d4b14

Reply:
It's for `DEFAULT_DOCKER_HOST` - it can't panic, because we know the value in advance.

But in the highlighted code the host might be a user's input

Reply:
You're right! Thanks for pointing it out.